python/ray/tune/schedulers/pbt.py

```diff
@@ -227,9 +244,32 @@ def __init__(self,
         self._num_checkpoints = 0
         self._num_perturbations = 0
 
+    def fill_config(self, config, attr, search_space):
+        # If attribute in hyperparam_mutations but not in config, add to config
+        if callable(search_space):
+            config[attr] = search_space()
+        elif isinstance(search_space, sample_from):
+            config[attr] = search_space.func(None)
+        elif isinstance(search_space, list):
+            config[attr] = random.choice(search_space)
+        elif isinstance(search_space, dict):
+            config[attr] = {}
+            for k, v in search_space.items():
+                self.fill_config(config[attr], k, v)
+
     def on_trial_add(self, trial_runner, trial):
         self._trial_state[trial] = PBTTrialState(trial)
 
+        for attr in self._hyperparam_mutations.keys():
+            if attr not in trial.config:
+                if log_once(attr + "-missing"):
+                    logger.info("Cannot find {} in config. Using search "
+                                "space provided by hyperparam_mutations.")
+                self.fill_config(trial.config, attr,
+                                 self._hyperparam_mutations[attr])
+            # Make sure this attribute is added to CLI output.
+            trial.evaluated_params[attr] = trial.config[attr]
```

Comment:
@richardliaw is this the best way to make sure these hparams are being shown in the CLI output? The current tests are failing because `MockTrial` does not have a `evaluated_params` attribute. I want to know if I should just change the tests, or we actually have to consider cases where trials don't have `evaluated_params` attribute,

Reply:
all trials should have evaluated_params, so its ok to change test.